pylint/test/functional/consider_using_get.py

```diff
@@ -0,0 +1,70 @@
+# pylint: disable=missing-docstring,invalid-name,using-constant-test,invalid-sequence-index,undefined-variable
+dictionary = dict()
+key = 'key'
+
+if 'key' in dictionary:  # [consider-using-get]
+    variable = dictionary['key']
+
+if 'key' in dictionary:  # [consider-using-get]
+    variable = dictionary['key']
+else:
+    variable = 'default'
+
+if key in dictionary:  # [consider-using-get]
+    variable = dictionary[key]
+
+if 'key' in dictionary:  # not accessing the dictionary in assignment
+    variable = "string"
+
+if 'key' in dictionary:  # is a match, but not obvious and we ignore it for now
+    variable = dictionary[key]
+
+if 'key1' in dictionary:  # dictionary querried for wrong key
+    variable = dictionary['key2']
+
+if 'key' in dictionary:  # body is not pure
+    variable = dictionary['key']
+    print('found')
+
+if 'key' in dictionary:  # body is not pure
+    variable = dictionary['key']
+    print('found')
+else:
+    variable = 'default'
+
+if 'key' in dictionary1:  # different dictionaries
+    variable = dictionary2['key']
+else:
+    variable = 'default'
+
+if 'key' in dictionary:  # body is not pure
+    variable = dictionary['key']
+else:
+    variable = 'default'
+    print('found')
+
+if 'key' in dictionary:  # different variables
+    variable1 = dictionary['key']
+else:
+    variable2 = 'default'
+
+if 'key' in dictionary:  # assignment is not simple
```

Comment:
We might also want to add a test for a complex Compare node, for which we shouldn't emit the error, say something along the lines of `if key in dict and something_else_about(key)...`.

Reply:
Good idea. Will do